source/extensions/transport_sockets/alts/tsi_socket.cc

```diff
@@ -106,14 +106,20 @@ Network::PostIoAction TsiSocket::doHandshakeNextDone(NextResultPtr&& next_result
     }
     if (handshake_validator_) {
       std::string err;
-      const bool peer_validated = handshake_validator_(peer, err);
+      TsiInfo tsi_info;
+      const bool peer_validated = handshake_validator_(peer, tsi_info, err);
```

Comment:
Would it be possible to add tests to validate the functional changes introduced by this PR?

Reply:
yes, I am still trying to figure out how to verify the dynamic metadata is actually set.